doc/03_reference/coverage_plan.rst

```diff
@@ -0,0 +1,299 @@
+* Microarchitectural coverage - which is concerned with the specifics of the RTL operation, ensuring interesting corner cases are seen along with various micro-architectural events (e.g. the different kinds of stall) and combinations of them.
+
+Architectural coverage is not Ibex specific and can be determined directly from a trace of executed instructions and is handled by RISC-V DV, details can be found in the `RISC-V DV documentation <https://htmlpreview.github.io/?https://github.com/google/riscv-dv/blob/master/docs/build/singlehtml/index.html#document-coverage_model>`_.
+
+Microarchitectural coverage will probe the Ibex RTL directly and is described here.
+There is some inevitable overlap between architectural and microarchitectural coverage but we aim to minimise it.
+
+Microarchitectural Events and Behaviour
+---------------------------------------
+
+Below are lists of specific things from the microarchitecture that will be included in functional coverage.
+Each of the points listed below must be covered.
+This will be further combined using cross coverage which is described in the section below.
+
+Instructions
+^^^^^^^^^^^^
+
+Categories
+""""""""""
+Instructions can be grouped into a number of categories.
+Each category represents a different kind of execution.
+For example the ``ADD`` and ``SUB`` instructions are in the same category as they are almost identical for the microarchitecture (both read two registers and write to one, both feed operands to the ALU and take their result from it, both have the same response to interrupts etc, the only difference is the ALU operation).
+
+Instructions can be compressed or uncompressed but that isn’t factored into the instruction categories below (excepting for illegal instructions).
+The decompress occurs in the IF stage and is invisible to the ID/EX stage so isn’t relevant for instruction execution.
+A separate set of category agnostic compressed instruction behaviour is considered instead.
+
+An instruction category is sampled at the ID/EX stage (which is where all the varying behaviours actually occur).
+Some categories are just a single instruction which is named without further description.
+
+
+* ALU - All of the reg/reg reg/imm instructions that use the ALU.
+  This is any RISC-V instruction with an opcode of ``7’b0010011`` or ``7’b0110011`` (``ibex_pkg::OPCODE_OP`` and ``ibex_pkg::OPCODE_OP_IMM``) other than the ``MUL*`` and ``DIV*`` family of instructions (from RV32M).
+* Mul - Any ``MUL*`` instruction (from RV32M).
+* Div - Any ``DIV*`` instruction (from RV32M).
+* Branch - Any ``B*`` family branch instruction.
+* Jump - ``JAL``/``JALR``
+* Load - Any ``L*`` family load instruction.
+* Store - Any ``S*`` family load instruction.
+* CSRAccess - Any instruction from Zicsr.
+* EBreakDbg/EBreakExc- An ``EBREAK`` instruction that either enters debug mode (Dbg) or causes an exception (Exc).
```

Comment:
Nit: Missing space before the hyphen. That said, if you're happy to distinguish the headings with a font change (as suggested above), a colon might be enough to separate things?